### PR TITLE
Add dependency bundles

### LINKED ISSUE #255: `notebook` dependency
(I'm a reviewer for the submission of `preliz` at JOSS)

This is in relation to the review criterion "Installation instructions: Is there a clearly-stated list of dependencies?".

Given the likely use of a jupyter notebook for the interactive aspects of the package, I'm wondering if `notebook` might be best listed as a dependency? Alternatively, perhaps it could be flagged in the "Installation" section of the docs that a `pip install notebook` may be required and again when `plot_interactive` is introduced.

### PR DESCRIPTION
**Pull request by aleicazatti**
## Description

* Add dependency bundles
* Update README and index
* Close  #255

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,11 +19,28 @@ The PreliZ documentation can be found in the [official docs](https://preliz.read
...
+```bash
+pip install "preliz[full]"
 ```
-pip install preliz
+
+:::
+:::{tab-item} Non-interactive
+
+```bash
+pip install "preliz[non-interactive]"
```
Comment (OriolAbril) [inline README.md:40]: This one should be `preliz`. The base set of dependencies (the ones specified [here](https://github.com/arviz-devs/preliz/blob/main/pyproject.toml#L25-L33)) are always installed. We can then also skip the `non-interactive` bundle.

File: README.md [hunk 2]
```diff
@@ -19,11 +19,28 @@ The PreliZ documentation can be found in the [official docs](https://preliz.read
...
 PreliZ is available for installation from [PyPI](https://pypi.org/project/preliz/).
-The latest  version can be installed using pip:
+There are two dependency bundles for the latest version:
 
+* `full`: includes dependencies for all features (interactive and non-interactive)
+* `non-interactive`: includes dependencies for non-interactive features only
+
+And both can be installed using pip:
+
+::::{tab-set}
```
Comment (OriolAbril) [inline README.md:29]: tabs do not work in the github readme nor on pypi, so I would not use them here, only in the docs (there is can be used from rst, md or ipynb files, but it is a sphinx feature)

File: pyproject.toml [hunk 3]
```diff
@@ -39,7 +39,23 @@ name = "preliz"
...
+  "ipywidgets",
+  "ipympl",
+  "jupyterlab",
+  "notebook",
+]
+non-interactive = [
+  "arviz",
+  "numpy>=1.22",
+  "scipy>=1.9.1",
+]
```
Comment (OriolAbril) [inline pyproject.toml:58]: Assuming we update the dependencies section (see comment above) to:

```
dependencies = [
  "arviz",
  "numpy>=1.22",
  "scipy>=1.9.1",
]
```

I would then update that to one of:

**opinionated version**: use jupyterlab

```
[project.optional-dependencies]
full = [
  "matplotlib>=3.5",
  "nbclient<0.6,>=0.2",
  "ipywidgets",
  "ipympl",
  "jupyterlab",
]
```

**ide agnostic version**:

```
[project.optional-dependencies]
full = [
  "matplotlib>=3.5",
  "nbclient<0.6,>=0.2",
  "ipywidgets",
  "ipympl",
]
lab = ["jupyterlab"]
notebook = ["notebook"]
```

which would modify install commands to:

```shell
pip install preliz # base
pip install "preliz[full,lab]"  # jupyterlab
pip install "preliz[full,notebook]"  # jupyter notebook
```

**Review by OriolAbril — COMMENTED** [review]
left some comments about how exactly we want to bundle the dependencies and optional tags. The current approach is a bit inconsistent. Later on we can discuss doc specifics, wordings and such

**Comment by OriolAbril** [comment]
I hope this will fix tests and docs, but we should probably set up a github action for testing "base" dependencies. As of now, the package might not be importable unless we do `preliz[full]`

File: .readthedocs.yaml [hunk 4]
```diff
@@ -3,12 +3,14 @@
...
-    python: "3.9"
+    python: "3.10"
 
 python:
    install:
    - requirements: requirements-docs.txt
    - method: pip
      path: .
+     extra_requirements:
+       - doc
```
Comment (OriolAbril) [inline .readthedocs.yaml:16]:
```suggestion
       - full
```

**Review by aloctavodia — COMMENTED** [review]
LGTM

File: pyproject.toml [hunk 5]
```diff
@@ -39,7 +35,15 @@ name = "preliz"
 source = "https://github.com/arviz-devs/preliz"
 tracker = "https://github.com/arviz-devs/preliz/issues"
 
-
+[project.optional-dependencies]
+full = [
+  "matplotlib>=3.5",
```
Comment (aloctavodia) [inline pyproject.toml:40]: Shouldn't matplotlib be listed as part of the regular dependencies? Also currently matplotlib will be installed anyway as we have arviz as a dependency

Comment (OriolAbril) [reply pyproject.toml:40]: Yes

**Pull request merged**